app/src/main/java/org/hackillinois/android/view/navigationdrawer/ScannerFragment.kt

```diff
@@ -0,0 +1,195 @@
+     */
+    fun processLastScan(lastScanWasSuccessful: Boolean?) {
+        when (lastScanWasSuccessful) {
+            false -> {
+                Toast.makeText(context, "Try again!", Toast.LENGTH_LONG).show()
+            }
+            true -> {
+                lastSuccessfullyScannedUser = viewModel.lastUserIdScannedIn.value.toString()
+                Toast.makeText(context, "Success: ${lastSuccessfullyScannedUser}",
+                        Toast.LENGTH_LONG).show()
+                staffOverrideSwitch.isChecked = false
+            }
+            null -> {
+                return
+            }
+        }
+
+    }
+
+    /**
+     * Extracts the userId from the HackIllinois QR String URI.
+     * Given an example URI "hackillinois://user?userid=github0000001"
+     * it is enough to grab everything after the last (and only) equals sign.
+     *
+     * If the URI becomes more complex, we can use actual URI decoders.
+     * @return the userId from a QR String
+     */
+    private fun getUserIdFromQrString(qrString: String): String {
+        var splitOnEquals = qrString.split("=")
+        return splitOnEquals.last()
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+
+        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                qrScanner.decodeContinuous(onQrCodeScan)
+            }
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate the layout for this fragment
+        var view = inflater.inflate(org.hackillinois.android.R.layout.fragment_scanner, container, false)
+
+        view.eventListView.onItemSelectedListener = viewModel.onEventSelected
+
+        // Ensure the Camera permission is granted
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
+                && activity!!.checkSelfPermission(Manifest.permission.CAMERA)
+                != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.CAMERA),
+                    PERMISSIONS_REQUEST_ACCESS_CAMERA)
+        } else {
+            view.qrScanner.decodeContinuous(onQrCodeScan)
+        }
+
+        return view
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+    }
+
+    /**
+     * Takes a list of events, grabs all of the name fields, creates an adapter with the list of
+     * names, and uses that to populate the Spinner.
+     * @param eventList the list of events from the latest publish event (local DB / API call)
+     */
+    private fun updateEventList(eventList: List<org.hackillinois.android.database.entity.Event>?) {
```

Comment:
This class name should just be "Event" and not "org.hackillinois.android.database.entity.Event"